solidity/contracts/KeepFactoryBeaconConsumer.sol

```diff
@@ -0,0 +1,144 @@
+  ▓▓▓▓▓▓▓▓▓▓▓▓▓▀        ▐▓▓▓▓▓▓▓▓▓▓         ▓▓▓▓▓▓▓▓▓▓▌        ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓
+  ▓▓▓▓▓▓▀▀▓▓▓▓▓▓▄       ▐▓▓▓▓▓▓▀▀▀▀         ▓▓▓▓▓▓▀▀▀▀         ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▀
+  ▓▓▓▓▓▓   ▀▓▓▓▓▓▓▄     ▐▓▓▓▓▓▓     ▓▓▓▓▓   ▓▓▓▓▓▓     ▓▓▓▓▓   ▐▓▓▓▓▓▌
+▓▓▓▓▓▓▓▓▓▓ █▓▓▓▓▓▓▓▓▓ ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓  ▓▓▓▓▓▓▓▓▓▓
+▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓ ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓  ▓▓▓▓▓▓▓▓▓▓
+
+                           Trust math, not hardware.
+*/
+
+pragma solidity 0.5.17;
+
+import "@keep-network/keep-core/contracts/IRandomBeacon.sol";
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+contract KeepFactoryBeaconConsumer is IRandomBeaconConsumer {
+    using SafeMath for uint256;
+
+    IRandomBeacon randomBeacon;
+
+    // Gas required for a callback from the random beacon. The value specifies
+    // gas required to call `__beaconCallback` function in the worst-case
+    // scenario with all the checks and maximum allowed uint256 relay entry as
+    // a callback parameter.
+    uint256 public constant callbackGas = 30000;
+
+    // Random beacon sends back callback surplus to the requestor. It may also
+    // decide to send additional request subsidy fee. What's more, it may happen
+    // that the beacon is busy and we will not refresh group selection seed from
+    // the beacon. We accumulate all funds received from the beacon in the
+    // reseed pool and later use this pool to reseed using a public reseed
+    // function on a manual request at any moment.
+    uint256 public reseedPool;
+
+    uint256 public groupSelectionSeed;
+
+    constructor(address _randomBeacon) public {
+        randomBeacon = IRandomBeacon(_randomBeacon);
+
+        // Initial value before the random beacon updates the seed.
+        // https://www.wolframalpha.com/input/?i=pi+to+78+digits
+        groupSelectionSeed = 31415926535897932384626433832795028841971693993751058209749445923078164062862;
+    }
+
+    /// @notice Adds any received funds to the reseed pool.
+    function() external payable {
+        reseedPool += msg.value;
+    }
+
+    /// @notice Sets a new group selection seed value.
+    /// @dev The function is expected to be called in a callback by the random
+    /// beacon.
+    /// @param _relayEntry Beacon output.
+    function __beaconCallback(uint256 _relayEntry) external onlyRandomBeacon {
+        groupSelectionSeed = _relayEntry;
+    }
+
+    /// @notice Gets a fee estimate for a new random entry.
+    /// @return Uint256 estimate.
+    function newEntryFeeEstimate() public view returns (uint256) {
+        return randomBeacon.entryFeeEstimate(callbackGas);
+    }
+
+    /// @notice Calculates the fee requestor has to pay to reseed the factory
+    /// for signer selection. Depending on how much value is stored in the
+    /// reseed pool and the price of a new relay entry, returned value may vary.
+    function newGroupSelectionSeedFee() public view returns (uint256) {
```

Comment:
I was thinking about renaming some of the functions to drop the relation to group selection to have this contract more generic. But this might require exposing functions with old names in the factory and many changes in docs and namings in this contract.

Reply:
I think the current naming is fine, it's not worth it. The only seeding/reseeding that may take place here is about group selection.